internal/cache/lru_cache.go

```diff
@@ -43,17 +47,22 @@ func NewLRUCache(capacity int) *LRUCache {
 	}
 }
 
-// GetElement gets an element from the cache.
+// GetElement gets an element from the cache. It returns
+// the associated data with the element with an error.
```

Comment:
Ive mentioned this is old PR

Reply:
Ref: Since we might not return owner always, I think we should abstract what it returns. It returns whatever we store in it. Makes it robust that way.

Reply:
Makes sense.